### ISSUE TITLE
Issue with Installing Nix After MacOS Update (Nix stopped working)

### ISSUE THREAD

**Issue by YashPotdar** [NixOS/nix#7880]
## Platform

- [ ] Linux: 
- [x] macOS
- [ ] WSL

## Output

<details><summary>Output</summary>
After updating MacOS software to Ventura 13.2, I cannot run `nix develop` anymore. I can a `nix: command not found` error. I tried running this `sh <(curl -L https://nixos.org/nix/install) --daemon` but run into an error:

I made sure to follow these steps below:
1. Back up (copy) /etc/bashrc and /etc/bashrc.backup-before-nix
   to another location, just in case.

2. Ensure /etc/bashrc.backup-before-nix does not have anything
   Nix-related in it. If it does, something is probably quite
   wrong. Please open an issue or get in touch immediately.
```log

mv /etc/bashrc.backup-before-nix /etc/bashrc
override r--r--r--  root/wheel for /etc/bashrc? (y/n [n]) y
mv: rename /etc/bashrc.backup-before-nix to /etc/bashrc: Permission denied

```

</details>

## Priorities

Add :+1: to [issues you find important](https://github.com/NixOS/nix/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by abathur** [comment]
For future reference: you don't have to reinstall after the update. The update is overwriting the shell hook, but you can just replace it. See #3616 for more.

At this point you'll have to follow the uninstall instructions before reinstalling it: https://nixos.org/manual/nix/stable/installation/installing-binary.html#macos

**Comment by Nicceboy** [comment]
> For future reference: you don't have to reinstall after the update. The update is overwriting the shell hook, but you can just replace it. See #3616 for more.

This update broke something more.
Whole daemon disappeared.
```sh
$ find . /nix/var/nix/profiles/ | grep daemon
$ 
```
Also my `~/.nix-profile` is empty directory.  `bin` directories have disappeared and I can't find them.

**Comment by abathur** [comment]
@Nicceboy Not sure what update you're referring to, but I doubt a macOS update removed any of these. 

I noticed some chatter about a regression related to profile paths in the 2.14.0 release of Nix (released yesterday). For reference, by ~chatter I mean:
- #7920
- #7925 
- #7929
- and discussion in #5588 starting at https://github.com/NixOS/nix/pull/5588#issuecomment-1448726090

I haven't had a chance to understand what happened there yet (and whether there's a quick workaround or when to expect a resolution), but I imagine it's what you're running into. 

For now, I'd recommend following the uninstall directions I mentioned in my previous comment, and then reinstall with 2.13.3 (use `https://releases.nixos.org/nix/nix-2.13.3/install` instead of `https://nixos.org/nix/install`) to see if that fixes whatever you're running into.

**Comment by Nicceboy** [comment]
That might be a case.

However, I just updated to the most recent MacOS 13.2.1 and everything broke afterwards. Everything worked before that.
Maybe I have updated Nix without noticing it and reboot changed something.

**Comment by abathur** [comment]
As noted in my response to the OP, macOS updates are known to overwrite the shell hook in /etc/zshrc, which will remove Nix from your PATH. 

I say that I doubt the macOS update removed any of the other bits because we occasionally get reports from people asserting that one macOS update or another overwrote/removed something else that Nix adds. Every time I see one of these, I go ask on Matrix to see if anyone else who has taken the same update observed the same problem, and I don't think I've been able to corroborate any except for /etc/zshrc. So, it's not impossible--but we'd need to know what's missing/changed, and we'd need enough reports from other users to be sure the update did it.

Your description might mean that your Nix Store volume has been deleted or isn't mounted, but I'm not really sure. `diskutil list` and `diskutil info /nix` could help clarify that. If you don't see the volume there, it might help to know if you have an /etc/synthetic.conf and /etc/fstab and what each contains. This could also be caused by a lot of other things, like encrypting your Nix Store after installing Nix or disabling the mounting service Nix installs. 

Happy to work through collecting information about what may be going on if you aren't in a hurry--but it could take a bit to figure out, and the recommendation (a full uninstall and reinstall of 2.13.3) is focused on getting you up and running.

**Comment by Nicceboy** [comment]
Yeah, I know that macOS has very unlikely removed these files itself, is just wondered if it has touched Nix in some way which caused it to lose symlinks.

As update, I reinstalled the latest version and the problem persisted.

I reinstalled with the suggested version and it works as used to. The problem seems to be in the  latest Nix version.

**Comment by ekaeoq** [comment]
Has anyone gotten around to fixing that issue? Just installed Sonoma and everything broke again, reinstalling the whole thing every time I update is so tedious

**Comment by Nicceboy** [comment]
Append following to the `/etc/zshrc`
```sh
# Nix
if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
  . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
fi
# End Nix
```

**Comment by mpstaton** [comment]
> point

The link you sent has uninstall instructions not install instructions.

**Comment by abathur** [comment]
> > point
> 
> The link you sent has uninstall instructions not install instructions.

The documentation's been reorganized in the 21 months since my comment. I pointed out that it broke the links at the time, but I guess it was changed in such a way that the anchors couldn't readily be remapped.

Here's my current version of the same saved reply:

You'll have to follow the uninstall instructions before reinstalling it: https://nixos.org/manual/nix/stable/installation/uninstall.html#macos

**Comment by mpstaton** [comment]
This is what happens when I follow the uninstall instructions:
<img width="572" alt="image" src="https://github.com/user-attachments/assets/15ee7864-21e4-405a-b353-56cfbbff04c3">

**Comment by abathur** [comment]
You may need to kill that process or try rebooting.

**Comment by lhohan** [comment]
> Append following to the `/etc/zshrc` after updating.
> 
> ```shell
> # Nix
> if [ -e '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh' ]; then
>   . '/nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh'
> fi
> # End Nix
> ```

Maybe it helps someone arriving here same way I did,
This worked for me when upgrading to Sequoia 15.2.
